andela/ah-shakas
# #160577484  Follow and Unfollow User

#### What does this PR do?
* Have users follow and unfollow each other according to their preferences in Authors' Haven
* A user will also be able to know who they are following or who is following them
#### Description of Task to be completed?
Have the following endpoints working:
*  POST /api/profile/{username}/follow
*  DELETE /api/profile/{username}/follow
*  GET /api/profile/{username}/following/
*  GET api/profile/{username}/followers/

#### How should this be manually tested?
* Clone this repository from [here](https://github.com/andela/ah-shakas)
* cd into the repository and run `git checkout-follow-user-160577484`
* Install requirements by running `pipenv install`
* activate virtual environment by running `pipenv shell`
* run migrations by running command `python manage.py migrate`
* Start the project by running `python manage.py runserver` 
* Using postman,create a user through the endpoint `/api/users/`
* payload for create user is 
              `{
              "user":{
              "username":"lala",
              "email": "aleni@jake.jake",
              "password": "jakejake1"
                       }
                }`
* login at endpoint `/api/users/login/` using the same payload as above
* Follow user at endpoint POST `/api/profile/{username}/follow` the username you provide should be 
   of a user who already exists in the system.
* To unfollow a user, do it at endpoint DELETE `/api/profile/{username}/follow`.
* To get the people you are following, go to endpoint GET `/api/profile/{username}/following/`.
* To get your followers, go the endpoint GET `api/profile/{username}/followers/`.
#### Any background context you want to provide?
 * A user should be authenticated before you test.
#### What are the relevant pivotal tracker stories?
* [#160577479](https://www.pivotaltracker.com/n/projects/2198533/stories/160577479)
#### Screenshots (if appropriate)
screenshot for follow
![image](https://user-images.githubusercontent.com/23398223/46736942-f7440b00-cca2-11e8-8f61-c6323cdc783d.png)
